homeassistant/components/binary_sensor/nissan_leaf.py

```diff
@@ -0,0 +1,49 @@
+"""
+Plugged In Status Support for the Nissan Leaf
+
+Documentation pending.
+Please refer to the main platform component for configuration details
+"""
+
+import logging
+
+from .. import nissan_leaf as LeafCore
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['nissan_leaf']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    devices = []
+

```

Comment:
You will have to add 

```python
if discovery_info is not None:
    return
```

to all platforms because the component is triggering a load of the platforms. Otherwise you will get issues when a user would add the leaf as a binary sensor platform.